analyzer/src/main/kotlin/managers/Carthage.kt

```diff
@@ -0,0 +1,296 @@
+        val projectUrl = "https://github.com/$id"
+        val vcsInfo = VcsInfo(
+            type = VcsType.GIT,
+            url = "$projectUrl.git",
+            revision = revision
+        )
+
+        return Package(
+            id = Identifier(
+                type = managerName,
+                namespace = VcsHost.GITHUB.getUserOrOrganization(projectUrl).orEmpty(),
+                name = VcsHost.GITHUB.getProject(projectUrl).orEmpty(),
+                version = revision
+            ),
+            declaredLicenses = sortedSetOf(),
+            description = "",
+            homepageUrl = projectUrl,
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = RemoteArtifact(
+                url = VcsHost.GITHUB.getArchiveDownloadUrl(vcsInfo).orEmpty(),
+                hash = Hash.NONE
+            ),
+            vcs = vcsInfo
+        )
+    }
+
+    private fun createPackageFromLocalRepository(fileUrl: String, revision: String): Package {
+        val vcsInfo = VcsInfo(
+            type = VcsType.GIT,
+            url = fileUrl,
+            revision = revision
+        )
+
+        return Package(
+            id = Identifier(
+                type = managerName,
+                namespace = "",
+                name = fileUrl.substringAfterLast("/"),
+                version = revision
+            ),
+            declaredLicenses = sortedSetOf(),
+            description = "",
+            homepageUrl = "",
+            binaryArtifact = RemoteArtifact.EMPTY,
+            sourceArtifact = RemoteArtifact.EMPTY,
+            vcs = vcsInfo
+        )
+    }
+
+    private fun createPackageFromBinarySpec(binarySpec: Map<String, String>, id: String, revision: String) = Package(
+        id = Identifier(
+            type = managerName,
+            namespace = "",
+            name = id.substringAfterLast("/").removeSuffix(".json"),
+            version = revision
+        ),
+        declaredLicenses = sortedSetOf(),
+        description = "",
+        homepageUrl = "",
+
+        // The content from the artifactUrl might contain header files that should be scanned.
+        // Therefore adding this URL for both sourceArtifact and binaryArtifact.
+        binaryArtifact = RemoteArtifact(
+            url = binarySpec[revision].orEmpty(),
+            hash = Hash.NONE
+        ),
+        sourceArtifact = RemoteArtifact(
+            url = binarySpec[revision].orEmpty(),
+            hash = Hash.NONE
+        ),
+        vcs = VcsInfo.EMPTY
+    )
+
+    private fun isGithubUserAndProject(id: String): Boolean {
+        val regex = Regex("([a-zA-Z0-9]+)/([a-zA-Z0-9]+)")
+        val match = regex.matchEntire(id)
+
+        return match != null
+    }
+
+    private fun isAbsolutePath(filePath: String) = filePath.startsWith("/")
+
+    private fun isFilePath(workingDir: String, path: String) =
+        path.startsWith("file://") || File(path).exists() || File("$workingDir/$path").exists()
```

Comment:
If the `File` has to be a directory, please use `isDirectory` instead of `exists()`.

Reply:
This method is used in two cases when it should return true:
1. The path to check is a local git repository and according to the Carthage Docs need to start with `file://`
2. The path is a local binary specification json. Either an absolute path or a relative path.

Using `isDirectory` would only work for the first case. Should I split this and use another method per use case?

Reply:
No splitting necessary IMO, but what you just described above probably deserves a code comment.

Reply:
IMO what you replied to me above is better than the docs you added. The docs just repeat what is obvious from the code, but your comment gave background information *why* that implementation makes sense. While I agree that such docs should probably not go to the functions docs, what I was meaning was in fact a code comment (not function docs) that should go between line 244 and 245, like:

    private fun isFilePath(workingDir: String, path: String) =
        // This covers the two cases supported Carthage, where either the path start with "file://" and points to a
        // local Git repository, or the path is a local binary specification JSON.
        path.startsWith("file://") || File(path).exists() || File("$workingDir/$path").exists()